table/tables/index.go

```diff
@@ -213,7 +217,11 @@ func (c *index) Create(sctx sessionctx.Context, txn kv.Transaction, indexedValue
 	}
 	if err != nil || len(value) == 0 {
 		if sctx.GetSessionVars().LazyCheckKeyNotExists() && err != nil {
-			err = txn.GetMemBuffer().SetWithFlags(key, idxVal, kv.SetPresumeKeyNotExists)
+			if opt.NewlyInserted {
+				err = txn.GetMemBuffer().SetWithFlags(key, idxVal, kv.SetPresumeKeyNotExists, kv.SetNewlyInserted)
+			} else {
+				err = txn.GetMemBuffer().SetWithFlags(key, idxVal, kv.SetPresumeKeyNotExists)
+			}
```

Comment:
If in such a table:
```sql
create table t (id int primary key, v int, unique index (v));
insert into t values (1, 10);
```
If we  execute:
```sql
begin;
delete from t where id = 1;
insert into t values (2, 10);
delete from t where id = 2;
commit;
```
Is it possible that the unique key is regarded as a delete-your-write and does noop on this transaction, results in the index key being left not deleted?

Reply:
The `newlyInserted` flag is used with `presumeNotExist` flag which ensures the targe key does not exist both in `memdb` and the store. If the unique key is deleted first, the get operation in `memdb` will return zero length value and a nil error. So the flag will not use for this case.